pkg/agent/baker.go

```diff
@@ -578,6 +578,18 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			}
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntimeConfig[datamodel.ContainerDataDirKey]
 		},
+		"HasKubeletDiskType": func() bool {
+			if profile != nil && profile.KubeletDiskType != "" {
```

Comment:
can we move kubeletdisktype to the NodeBootstrapConfiguration? We want to get rid of agent pool profile

Reply:
maybe in the next PR if you need to rush this in

Reply:
also return profile != nil && profile.KubeletDiskType != ""